js/widgets/modewidget.js

```diff
@@ -9,152 +9,145 @@
-        this.widgetWindow = widgetWindow;
-        widgetWindow.clear();
-	widgetWindow.show();
+        const iconSize = this.CONSTANTS.ICONSIZE * this._cellScale;
 
-        // For the button callbacks
-        let that = this;
+        this.widgetWindow = window.widgetWindows.windowFor(this, "custom mode");
+        this.widgetWindow.clear();
+	    this.widgetWindow.show();
 
         this.modeTableDiv = document.createElement("div");
-        widgetWindow.getWidgetBody().append(this.modeTableDiv);
+        this.widgetWindow.getWidgetBody().append(this.modeTableDiv);
 
-        widgetWindow.onclose = function() {
-            that._logo.hideMsgs();
-            this.destroy();
+        this.widgetWindow.onclose = ()=>{
+            this._logo.hideMsgs();
+            this.widgetWindow.destroy();
         };
 
-        this._playButton = widgetWindow.addButton(
+        this._playButton = this.widgetWindow.addButton(
             "play-button.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Play")
         );
-        this._playButton.onclick = function() {
-            that._logo.resetSynth(0);
-            if (that._playingStatus()) {
-                that._playing = false;
+        this._playButton.onclick = ()=>{
+            this._logo.resetSynth(0);
+            if (this._playingStatus()) {
+                this._playing = false;
 
-                this.innerHTML =
+                this._playButton.innerHTML =
                     '&nbsp;&nbsp;<img src="header-icons/play-button.svg" title="' +
                     _("Play all") +
                     '" alt="' +
                     _("Play all") +
                     '" height="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" width="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" vertical-align="middle">&nbsp;&nbsp;';
             } else {
-                that._playing = true;
+                this._playing = true;
 
-                this.innerHTML =
+                this._playButton.innerHTML =
                     '&nbsp;&nbsp;<img src="header-icons/stop-button.svg" title="' +
                     _("Stop") +
                     '" alt="' +
                     _("Stop") +
                     '" height="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" width="' +
-                    ICONSIZE +
+                    this.CONSTANTS.ICONSIZE +
                     '" vertical-align="middle">&nbsp;&nbsp;';
 
-                that._playAll();
+                this._playAll();
             }
         };
 
-        widgetWindow.addButton(
+        this.widgetWindow.addButton(
             "export-chunk.svg",
-            ICONSIZE,
+            this.CONSTANTS.ICONSIZE,
             _("Save")
-        ).onclick = function() {
-            that._save();
+        ).onclick = ()=>{
```

Comment:
`= foo` better than `= () => { foo(); }` IMHO ... it is a callback in any case.

But `= () => foo();` might be easier to follow IG ... decreases a couple of silly lines , and still identifies it as a function.